### PR TITLE
NXP-25378: Use trash state computation in TrashService

### PR DESCRIPTION
**Pull request by pierre-gautier**
https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-pierregautier3/10/

### CODE REVIEW

File: nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/RemoveProxyAction.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.IterableQueryResult;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+import org.nuxeo.lib.stream.computation.Topology.Builder;
+
+/**
+ * Removes documents which are proxies and whose id is contained in the given ids list.
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/RemoveProxyAction.java]: Can you describe stream inputs/outputs ?

File: nuxeo-core/nuxeo-core-bulk/src/test/java/org/nuxeo/ecm/core/bulk/CoreBulkFeature.java [hunk 2]
```diff
@@ -47,25 +46,16 @@
...
-            return bulks.await(duration);
-        }
-
-    }
-
     @Override
     public void initialize(FeaturesRunner runner) {
-        runner.getFeature(TransactionalFeature.class).addWaiter(new BulkWaiter());
+        runner.getFeature(TransactionalFeature.class)
+              .addWaiter(deadline -> Framework.getService(BulkService.class).await(Duration.ofMinutes(1)));
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-bulk/src/test/java/org/nuxeo/ecm/core/bulk/CoreBulkFeature.java]: You're changing logic there, duration becomes deadline which is no more taken into account.

File: nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/TestTrashBackwardMechanism.java [hunk 3]
```diff
@@ -120,12 +116,10 @@ public void testMechanismTriggersOneListener() {
...
     public static class BulkListenersFilter implements LogCaptureFeature.Filter {
 
-        protected final Set<String> bulkListenerNames = Stream.of(BulkLifeCycleChangeListener.class,
-                BulkTrashedStateChangeListener.class).map(Class::getName).collect(Collectors.toSet());
-
         @Override
         public boolean accept(LoggingEvent event) {
-            return Level.DEBUG.equals(event.getLevel()) && bulkListenerNames.contains(event.getLoggerName());
+            return Level.DEBUG.equals(event.getLevel())
+                    && event.getLoggerName().equals(BulkLifeCycleChangeListener.class.getName());
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/TestTrashBackwardMechanism.java]: Filter can now be converted to `FilterOn` annotation.

Comment (kevinleturc) [reply nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/TestTrashBackwardMechanism.java]: Also you're changing logic of test which asserted that only one recursive mechanism was trigger.

File: nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestTreeIndexing.java [hunk 4]
```diff
@@ -398,6 +403,7 @@ public void shouldStoreOnlyEffectiveACEs() throws Exception {
     }
 
     @Test
+    @Deploy("org.nuxeo.elasticsearch.core.test:dummy-bulk-login-config.xml")
```
Comment (kevinleturc) [inline nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestTreeIndexing.java]: If it's the same than one present in bulk or bulk test, you can reuse it as test-jar should be in classpath.

Comment (pierre-gautier) [reply nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestTreeIndexing.java]: Actually it's not the same, it has to be deployed later than the one in bulk and has a different component name, it sucks but I could not find any other way, any bright idea ?

**Comment by pierre-gautier** [comment]
https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-pierregautier/

**Comment by pierre-gautier** [comment]
PR is closed since it will be heavily reworked after being rebased on further work on bulk service implementation
https://jira.nuxeo.com/browse/NXP-25940

**Pull request closed without merging**